deps/verifier/src/eventlog/mod.rs

```diff
@@ -1,187 +0,0 @@
-    }
-}
-
-#[derive(Clone)]
-pub struct AAEventlog {
-    pub hash_algorithm: HashAlgorithm,
-    pub init_state: Vec<u8>,
-    pub events: Vec<AAEvent>,
-}
-
-impl FromStr for AAEventlog {
-    type Err = anyhow::Error;
-
-    fn from_str(input: &str) -> Result<Self> {
-        let all_lines = input.lines().collect::<Vec<&str>>();
-
-        let (initline, eventlines) = all_lines
-            .split_first()
-            .ok_or(anyhow!("at least one line should be included in AAEL"))?;
-
-        // Init line looks like
-        // INIT sha256/0000000000000000000000000000000000000000000000000000000000000000
-        let init_line_items = initline.split_ascii_whitespace().collect::<Vec<&str>>();
-        if init_line_items.len() != 2 {
-            bail!("Illegal INIT event record.");
-        }
-
-        if init_line_items[0] != "INIT" {
-            bail!("INIT event should start with `INIT` key word");
-        }
-
-        let (hash_algorithm, init_state) = init_line_items[1].split_once('/').ok_or(anyhow!(
-            "INIT event should have `<sha-algorithm>/<init-PCR-value>` as content after `INIT`"
-        ))?;
-
-        let hash_algorithm = hash_algorithm
-            .try_into()
-            .context("parse Hash Algorithm in INIT entry")?;
-        let init_state = hex::decode(init_state).context("parse init state in INIT entry")?;
-
-        let events = eventlines
-            .iter()
-            .map(|line| AAEvent::from_str(line))
-            .collect::<Result<Vec<AAEvent>>>()?;
-
-        Ok(Self {
-            events,
-            hash_algorithm,
-            init_state,
-        })
-    }
-}
-
-impl AAEventlog {
-    fn accumulate_hash<D: Digest + FixedOutput>(&self) -> Vec<u8> {
-        let mut state = self.init_state.clone();
-
-        let mut init_event_hasher = D::new();
-        let init_event = format!(
-            "INIT {}/{}",
-            self.hash_algorithm.as_ref(),
-            hex::encode(&self.init_state)
-        );
-        Digest::update(&mut init_event_hasher, init_event.as_bytes());
-        let init_event_hash = init_event_hasher.finalize();
-
-        let mut hasher = D::new();
-        Digest::update(&mut hasher, &state);
-
-        Digest::update(&mut hasher, init_event_hash);
-        state = hasher.finalize().to_vec();
-
-        self.events.iter().for_each(|event| {
-            let mut event_hasher = D::new();
-            Digest::update(&mut event_hasher, event.domain.as_bytes());
-            Digest::update(&mut event_hasher, b" ");
-            Digest::update(&mut event_hasher, event.operation.as_bytes());
-            Digest::update(&mut event_hasher, b" ");
-            Digest::update(&mut event_hasher, event.content.as_bytes());
-            let event_hash = event_hasher.finalize();
-
-            let mut hasher = D::new();
-            Digest::update(&mut hasher, &state);
-            Digest::update(&mut hasher, event_hash);
-            state = hasher.finalize().to_vec();
-        });
```

Comment:
this one had one important feature we don't have with in the TCG2 EV_EVENT_TAG parser: we don't provide the guarantees that the AAEL content is correctly included in the hash and matches the event description.

Reply:
> hashing/recording work is responsibility of tcg2 eventlog

what I wanted to say was that these new claims:

```json
"data": {
                    "content": "some-content",
                    "domain": "coco.com",
                    "operation": "some-operation"
                },
                "string": "coco.com some-operation some-content",
               ...
}
```

alone cannot be trusted because we don't check `digest == sha384("coco.com some-operation some-content"). but if these are just hints and the real trust decision is only done based on the digest reference value, we are ok.

the old AAEL integrity check verified the line content also.

Reply:
they will be checked as EV_TAG_EVENT are fully hashed

Reply:
I don't see where we verity the event data matches the hash that was included.

Reply:
Thanks, that is enough for know.